satpy/tests/test_readers.py

```diff
@@ -439,6 +444,24 @@ def test_almost_all_filtered(self):
         assert "abi_l1b" in readers
         assert len(list(readers["abi_l1b"].available_dataset_ids)) == 0
 
+    @mock.patch("satpy.readers.load_reader")
```

Comment:
Would it be possible to test without mocking? eg having a test yaml file and a test python file that imports garbage?

Reply:
I'm trying, lets see if I manage in this but seems to be quite complicated.

Reply:
https://github.com/pytroll/satpy/pull/2931/commits/3db654715280240f8fbc00690811e3350106bb7c replaces the mock with a reader config file that tries to use a non-existent module triggering the `yaml.constructor.ConstructorError`.